models/page.js

```diff
@@ -17,6 +17,16 @@ let pageSchema = new Schema({
     owners: Array,
     editors: Array,
     viewers: Array
+});
+
+pageSchema.pre('save', function (next) {
```

Comment:
What is this for?

Reply:
it set a pre function, ie I makes the schema then before saving it runs that method. that's where I set filename and page_id if they don't exist.